### PR TITLE
Collapse Client.ReadKV() into Client.Read()

### PR DESCRIPTION
**Pull request by benashz**
Unify client.ReadKV() with Client.Read() by implementing some common response/request interfaces. This means one less method on the Client interface, which should make overall Client interface simpler.

### CODE REVIEW

File: internal/vault/responses.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import "github.com/hashicorp/vault/api"
+
+var (
+	_ Response = (*defaultResponse)(nil)
+	_ Response = (*kvV2Response)(nil)
+)
+
+type Response interface {
+	Secret() *api.Secret
+	Data() map[string]interface{}
```
Comment (tvoran) [inline internal/vault/responses.go:15]: Is there anything preventing this from returning `map[string]any` instead?

File: controllers/vaultdynamicsecret_controller.go [hunk 2]
```diff
@@ -293,9 +293,9 @@ func (r *VaultDynamicSecretReconciler) syncSecret(ctx context.Context, c vault.C
 
 	switch method {
 	case http.MethodPut, http.MethodPost:
-		resp, err = c.Write(ctx, path, params)
+		resp, err = c.Write(ctx, vault.NewWriteRequest(path, params, method))
```
Comment (tvoran) [inline controllers/vaultdynamicsecret_controller.go:296]: I'm curious how the `method` param is used? I'm not seeing `Method()` called in the client, for example.

Comment (benashz) [reply controllers/vaultdynamicsecret_controller.go:296]: Oops, yeah it is only related to the VDS controller. All Write() requests are `PUT`. Fixing.

File: internal/vault/responses.go (hunk 1 above)
Comment (benashz) [reply internal/vault/responses.go:15]: Nope, should be fine to do so.

Comment (benashz) [reply internal/vault/responses.go]: Updated with a74d5224b88c8929d26ee7380886c267a29d10aa

File: controllers/vaultdynamicsecret_controller.go (hunk 2 above)
Comment (benashz) [reply controllers/vaultdynamicsecret_controller.go]: Fixed with a74d5224b88c8929d26ee7380886c267a29d10aa

**Review by tvoran — APPROVED** [review]

**Pull request merged**